shiva/src/main/scala/nl/lumc/sasc/biopet/pipelines/shiva/ShivaSvCalling.scala

```diff
@@ -96,17 +97,54 @@ class ShivaSvCalling(val parent: Configurable) extends QScript with SummaryQScri
     // group by "tags"
     // sample tagging is however not available within this pipeline
 
+    for ((sample, mergedResultFile) <- outputMergedVCFbySample) {
+      lazy val counts = getVariantCounts(mergedResultFile, ShivaSvCallingReport.histogramBinBoundaries)
+      addSummarizable(new Summarizable {
+        def summaryFiles = Map("output_vcf" -> mergedResultFile)
+        def summaryStats = counts
+      }, "parse_sv_vcf", Some(sample))
+    }
+
     addSummaryJobs()
   }
 
   /** Will generate all available variantcallers */
   protected def callersList: List[SvCaller] = List(new Breakdancer(this), new Clever(this), new Delly(this), new Pindel(this))
 
   /** Settings for the summary */
-  def summarySettings = Map("sv_callers" -> configCallers.toList)
+  def summarySettings = Map("sv_callers" -> configCallers.toList, "hist_bin_boundaries" -> ShivaSvCallingReport.histogramBinBoundaries)
 
   /** Files for the summary */
-  def summaryFiles: Map[String, File] = Map("final_mergedvcf" -> (if (inputBams.size > 1) outputMergedVCF else outputMergedVCFbySample.values.head))
+  def summaryFiles: Map[String, File] = if (inputBams.size > 1) Map("final_mergedvcf" -> outputMergedVCF) else Map.empty
+
+  def getVariantCounts(vcfFile: File, breaks: Array[Int]): Map[String, Any] = {
+    val delCounts, insCounts, dupCounts, invCounts = Array.fill(breaks.size + 1) { 0 }
+    var traCount = 0
```

Comment:
Aha, I get it now. So `breaks` is an array of counts per SV size? If so, this then relies on `breaks` having a specific ordering of boundaries (that is not fully apparent at this point in the code).

What about specifically encoding the SV size boundaries? Something like this:

```
case class SvBoundary(lower: Long, upper: Long) {
    def containsSize(svSize: Long) = lower <= svSize && upper >= svSize
}
```

You would then use this case class's `containsSize` to match a given SV size.

The counts can then be stored in something like `collection.mutable.Map[SvBoundary, Int]`.